homework1/Lab2.js

```diff
@@ -0,0 +1,394 @@
+        this.render();
+    }
+
+    render () {
+        // TODO: render a list of food menu from store using innerHTML
+    // using innerHTML to render inventory listing
+            let tbody = this.root.querySelector('tbody');
+            // using innerHTML to render a list of table row item under tbody
+            tbody.innerHTML = ``;
+            if (this.store.foods === null) {
+                this.store.foods = [];
+            } 
+            for (var i = 0; i < this.store.foods.length; i++) {
+                // for each item in local storage's FOODS, create a row with a cell for the item name and image, and one for description.
+                var food_name = this.store.foods[i].name;
+                var food_image = this.store.foods[i].image;
+                var food_description = this.store.foods[i].description;
+
+                tbody.innerHTML +=
+                    `<tr >
+                     <td>${food_name}</td>
+                        <td>
+                            <img src=${food_image}>
+                        </td>
+                        <td>${food_description}
+                        </td>
+                        <td>
+                            <button class="checkout-button" data-index=${i}>Add item to cart</button>
+                         </td>
+                      </tr>`;
+            }
+
+            let checkoutButtons =document.querySelectorAll('.checkout-button');
+            if(checkoutButtons&& checkoutButtons.length){
+                for(var i=0;i<checkoutButtons.length;i++){
+                    new CheckoutButton(checkoutButtons[i], this.store,this.cart);
+                }
+            }
+
+
+    }
+}
+
+class CreateFood {
+    constructor(root, store) {
+        this.root = root; // root should be the container of the form itself
+        this.store = store;
+        this.init();
+    }
+
+    init () {
+        // attach click event to create button
+        let Add_foodbutton=document.getElementById('Add_food');
+        if(Add_foodbutton){
+            Add_foodbutton.addEventListener("click",()=>{
+
+                this.createFood();
+            }
+
+
+                )
+            document.getElementById("").innerHTML='';
+            
+        }
+    }
+
+    createFood () {
+        // will need to do querySelector to find out every single form element
+        // to get their values before creating a new food
+        // after creating a new food item, add it to store
+        let store_food=this.store.foods || [];
+        var food_name=document.getElementById('food_name');
+        var food_image=document.getElementById('food_img');
+        var description=document.getElementById('food_description');
+        var pushing={name:food_name,image:food_image,description:description};
+
+        var boolean=true; 
+        for(var i=0;i<store_food.length;i++){
+            if(pushing.name===store_food[i].name)
+                boolean=false;
+                break;
```

Comment:
Should surround the if statement with open "{" and close "}". Otherwise, only one line will be executed at the if statement. In other word, the break statement will always be executed if you don't use brackets.